addons/binding/org.openhab.binding.loxone/src/main/java/org/openhab/binding/loxone/core/LxWsClient.java

```diff
@@ -0,0 +1,765 @@
+                        disconnect();
+                    }
+                }
+            }, connectTimeout, TimeUnit.SECONDS);
+        }
+    }
+
+    /**
+     * Stops scheduled timeout waiting for a Miniserver response
+     */
+    private void stopResponseTimeout() {
+        logger.trace("[{}] stopping response timeout in state {}", debugId, state.toString());
+        synchronized (state) {
+            if (timeout != null) {
+                timeout.cancel(true);
+                timeout = null;
+            }
+        }
+    }
+
+    /**
+     * Sends an event to {@link LxServer} object
+     *
+     * @param event
+     *            event that happened
+     * @param reason
+     *            reason for the event (applicable to server OFFLINE event}
+     * @param object
+     *            additional data for the event (text message for OFFLINE event, data for state changes)
+     */
+    private void notifyMaster(EventType event, OfflineReason reason, Object object) {
+        if (reason == null) {
+            reason = OfflineReason.NONE;
+        }
+        LxServerEvent sync = new LxServerEvent(event, reason, object);
+        try {
+            queue.put(sync);
+        } catch (InterruptedException e) {
+            logger.debug("[{}] Interrupted queue operation", debugId);
+        }
+    }
+
+    /**
+     * Implementation of jetty websocket client
+     *
+     * @author Pawel Pieczul - initial commit
+     *
+     */
+    @WebSocket
+    public class LxWebSocket {
+        Session session;
+        private ScheduledFuture<?> keepAlive = null;
+        private LxWsBinaryHeader header = null;
+
+        @OnWebSocketConnect
+        public void onConnect(Session session) {
+            synchronized (state) {
+                if (state != ClientState.CONNECTING) {
+                    logger.debug("[{}] Unexpected connect received on websocket in state {}", debugId,
+                            state.toString());
+                    return;
+                }
+
+                WebSocketPolicy policy = session.getPolicy();
+                policy.setMaxBinaryMessageSize(maxBinMsgSize * 1024);
+                policy.setMaxTextMessageSize(maxTextMsgSize * 1024);
+
+                logger.debug("[{}] Websocket connected (maxBinMsgSize={}, maxTextMsgSize={})", debugId,
+                        policy.getMaxBinaryMessageSize(), policy.getMaxTextMessageSize());
+                this.session = session;
+                setClientState(ClientState.CONNECTED);
+
+                startResponseTimeout();
+
+                try {
+                    sendString(CMD_GET_KEY);
+                } catch (IOException e) {
+                    notifyAndClose(OfflineReason.COMMUNICATION_ERROR,
+                            "Message send failure on recently connected websocket");
+                }
+            }
+            keepAlive = SCHEDULER.scheduleAtFixedRate(new Runnable() {
```

Comment:
Please use [scheduleWithFixedDelay](http://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ScheduledExecutorService.html#scheduleWithFixedDelay(java.lang.Runnable,%20long,%20long,%20java.util.concurrent.TimeUnit)) which is preferred over scheduleAtFixedRate, see [Coding Guidelines D.3](http://docs.openhab.org/developers/development/guidelines.html#d-runtime-behavior)